helix-core/src/test/java/org/apache/helix/integration/task/TestForceDeleteWorkflow.java

```diff
@@ -0,0 +1,368 @@
+          && (job1 == null || job1 == TaskState.STOPPED)
+          && (job2 == null || job2 == TaskState.STOPPED));
+    }, 60 * 1000);
+    Assert.assertTrue(isWorkflowStopped);
+
+    // Check that WorkflowConfig, WorkflowContext, and IdealState are indeed created for this
+    // workflow.
+    Assert.assertNotNull(_driver.getWorkflowConfig(workflowName));
+    Assert.assertNotNull(_driver.getWorkflowContext(workflowName));
+    Assert.assertNotNull(_admin.getResourceIdealState(CLUSTER_NAME, workflowName));
+
+    // Stop the Controller
+    _controller.syncStop();
+
+    _driver.delete(workflowName, true);
+
+    // Check that WorkflowConfig, WorkflowContext, and IdealState are indeed deleted for this
+    // workflow.
+    boolean isWorkflowDeleted = TestHelper.verify(() -> {
+      WorkflowConfig wfcfg = _driver.getWorkflowConfig(workflowName);
+      WorkflowContext wfctx = _driver.getWorkflowContext(workflowName);
+      IdealState is = _admin.getResourceIdealState(CLUSTER_NAME, workflowName);
+      return (wfcfg == null && wfctx == null && is == null);
+    }, 60 * 1000);
+    Assert.assertTrue(isWorkflowDeleted);
+  }
+
+  @Test(dependsOnMethods = "testDeleteStoppedWorkflowForcefully")
+  public void testDeleteStoppingStuckWorkflowForcefully() throws Exception {
+    // In this test, Stuck workflow indicates a workflow that is in STOPPING state (user requested
+    // to stop the workflow), its JOBs are also in STOPPING state but the tasks are stuck (or taking
+    // a long time to stop) in RUNNING state.
+
+    // Reset the cancel and stop counts
+    CANCEL_COUNT.set(0);
+    STOP_COUNT.set(0);
+
+    // Start the Controller
+    String controllerName = CONTROLLER_PREFIX + "_0";
+    _controller = new ClusterControllerManager(ZK_ADDR, CLUSTER_NAME, controllerName);
+    _controller.syncStart();
+
+    String workflowName = TestHelper.getTestMethodName();
+    Workflow.Builder builder = createCustomWorkflow(workflowName, LONG_EXECUTION_TIME, STOP_DELAY);
+    _driver.start(builder.build());
+
+    // Wait until workflow is created and running.
+    _driver.pollForWorkflowState(workflowName, TaskState.IN_PROGRESS);
+
+    _driver.stop(workflowName);
+    // Wait until workflow is stopped.
+
+    // Wait until workflow is created and running.
+    _driver.pollForWorkflowState(workflowName, TaskState.STOPPING);
+
+    // Check the status of JOB0 to make sure it is running.
+    _driver.pollForJobState(workflowName, TaskUtil.getNamespacedJobName(workflowName, "JOB0"),
+        TaskState.STOPPING);
+
+    // Since ConcurrentTasksPerInstance is set to be 1, the total number of CANCEL_COUNT is expected
+    // to be equal to number of nodes
+    boolean haveAllCancelsCalled = TestHelper.verify(() -> {
+      return CANCEL_COUNT.get() == _numNodes;
+    }, 60 * 1000);
+    Assert.assertTrue(haveAllCancelsCalled);
+
+    Thread.sleep(1000);
+    Assert.assertEquals(STOP_COUNT.get(), 0);
```

Comment:
Explain?

"Check that STOP_COUNT is 0. This checks that no tasks' Task.cancel() have returned, verifying that they are indeed stuck."